python/lsst/ip/isr/isrFunctions.py

```diff
@@ -216,6 +217,113 @@ def growMasks(mask, radius=0, maskNameList=['BAD'], maskValue="BAD"):
+            # that have the same X coordinate as the saturated core.
+            # As we don't care about the Y coordinate, we set it to the
+            # center of the BBox.
+            yBox = amp.getBBox().getCenter()[1]
+            if amp.getBBox().contains(xCore, yBox):
+
+                # Get the amp name
+                ampName = amp.getName()
+
+                # Because in ITLs the edge bleed happens on both edges
+                # of the detector, we make a cutout around
+                # both the top and bottom
+                # edge bleed candidates around the saturated core.
+                # We flip the cutout of the top amplifier
+                # to then work with the same coordinates for both.
+                # The way of selecting top vs bottom amp
+                # is very specific to ITL.
+                if ampName[:2] == 'C1':
+                    sliceImage = maskedImage.image.array[:200, :]
+                    sliceMask = maskedImage.mask.array[:200, :]
+                elif ampName[:2] == 'C0':
+                    sliceImage = numpy.flipud(maskedImage.image.array[-200:, :])
+                    sliceMask = numpy.flipud(maskedImage.mask.array[-200:, :])
+
+                # The middle columns of edge bleeds are often quite close
+                # to the saturation level so
+                # we check there is an edge bleed by looking
+                # at a small image up to 50 pixels from the edge
+                # and around the saturated columns
+                # of the saturated core, and checking its mean is
+                # above 80 percent (which is set as default) of the
+                # saturation level.
+                if numpy.mean(sliceImage[:50, int(xCore)-5:int(xCore)+5]) \
+                        > itlEdgeBleedSatFracLevel*satLevel:
+
+                    # We need an estimate of the maximum width
+                    # of the edge bleed for our masking model
+                    # so we now estimate it by measuring the width of
+                    # areas above 60 percent of the saturation level
+                    # close to the edge,
+                    # in a cutout up to 100 pixels from the edge.
+                    subImage = sliceImage[:100, :]
+                    maxWidthEdgeBleed = numpy.max(numpy.sum(subImage > 0.6*satLevel,
+                                                            axis=1))
+
+                    # Mask edge bleed with a
+                    # decaying exponential model
+                    for y in range(200):
+                        edgeBleedHalfWidth = \
+                            int(((maxWidthEdgeBleed)*numpy.exp(-itlEdgeBleedModelConstant*y)
+                                 + widthSat)/2.)
+                        sliceMask[y, int(xCore)-edgeBleedHalfWidth:int(xCore)+edgeBleedHalfWidth] = \
+                            maskedImage.mask.getPlaneBitMask(saturatedMaskName)
```

Comment:
You want to take `maskedImage.mask.getPlaneBitMask(saturatedMaskName)` and bring it up and out of the loop and set it to a variable so we don't have to do the same operation hundreds of times (which can be surprisingly slow in some cases).